### PR TITLE
skeleton bus device for SNES copiers of the Pro Fighter family

### PR DESCRIPTION
**Pull request by Osso13**
- dumped 4 SNES copiers (3 Super Pro. Fighter Q variants and a Pro Fighter X) [Darksoft, Apocalypse]
- fixed PLD dumps to make them convertable via jedutil [Vas Crabb]
- created skeleton bus device for the Pro Fighter family of SNES copiers

### CODE REVIEW

File: src/devices/bus/snes/snes_carts.cpp [hunk 1]
```diff
@@ -66,4 +68,9 @@ void snes_cart(device_slot_interface &device)
 	device.option_add_internal("hirom_dsp1leg", SNS_HIROM_NECDSP1_LEG);
 	device.option_add_internal("lorom_st10leg", SNS_LOROM_SETA10_LEG);
 	device.option_add_internal("lorom_st11leg", SNS_LOROM_SETA11_LEG);
+	// copiers
+	device.option_add_internal("profighterq",   SNS_PRO_FIGHTER_Q);
+	device.option_add_internal("profighterqa",  SNS_PRO_FIGHTER_QA);
+	device.option_add_internal("profighterqb",  SNS_PRO_FIGHTER_QB);
+	device.option_add_internal("profighterx",   SNS_PRO_FIGHTER_X);
```
Comment (cuavas) [inline src/devices/bus/snes/snes_carts.cpp:75]: If you make these “internal” they’ll be impossible to use at all.

File: src/devices/bus/snes/profighter.cpp [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+const tiny_rom_entry *sns_pro_fighter_qb_device::device_rom_region() const
+{
+	return ROM_NAME( profghtqb );
+}
+
+void sns_pro_fighter_qb_device::device_add_mconfig(machine_config &config)
+{
+	N82077AA(config, "fdc", 16_MHz_XTAL); // actually ACCMicro 3201
+	FLOPPY_CONNECTOR(config, "fdc:0", profght_floppies, "35dd", floppy_image_device::default_pc_floppy_formats).enable_sound(true);
+}
```
Comment (cuavas) [inline src/devices/bus/snes/profighter.cpp:196]: You should probably add the cartridge slots to these devices – they have a SNES cartridge slot for connecting a game to copy, or a special cartridge providing a DSP chip for playing Pilot Wings, Mario Kart, etc.

File: src/devices/bus/snes/profighter.h [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+#ifndef MAME_BUS_SNES_PROFIGHTER_H
+#define MAME_BUS_SNES_PROFIGHTER_H
+
+#pragma once
+
+#include "snes_slot.h"
+
+#include "formats/pc_dsk.h"
+#include "imagedev/floppy.h"
+#include "machine/upd765.h"
```
Comment (cuavas) [inline src/devices/bus/snes/profighter.h:13]: Is there really a reason to have formats headers here?  It leaks downstream.

File: src/devices/bus/snes/profighter.h [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+class sns_pro_fighter_q_device : public device_t,
+						public device_sns_cart_interface
+{
+public:
+	// construction/destruction
+	sns_pro_fighter_q_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock);
+
+protected:
+	// device-level overrides
+	virtual void device_start() override;
```
Comment (cuavas) [inline src/devices/bus/snes/profighter.h:25]: Please don’t copy/paste this comment everywhere.  What does it even mean?  What‘s the device “level”?  What other “levels” are there?

**Comment by Osso13** [comment]
I've fixed my mindless copy/pasting + brain cramps.
As for adding the cart slots, I'd prefer to avoid it since my only intent here is provide a placeholder so the dumps don't get lost. Without further development this device does nothing anyway.

**Comment by DavidHaywood** [comment]
I think having placeholders for this kind of thing is important, because it's one of those things nobody will really care about these days

**Pull request merged**